instrumentation/hibernate/hibernate-4.0/javaagent/src/test/groovy/EntityManagerTest.groovy

```diff
@@ -0,0 +1,214 @@
+/*
+ * Copyright The OpenTelemetry Authors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+import static io.opentelemetry.api.trace.SpanKind.CLIENT
+import static io.opentelemetry.api.trace.SpanKind.INTERNAL
+import static io.opentelemetry.instrumentation.test.utils.TraceUtils.runUnderTrace
+
+import io.opentelemetry.semconv.trace.attributes.SemanticAttributes
+import javax.persistence.EntityManager
+import javax.persistence.EntityManagerFactory
+import javax.persistence.EntityTransaction
+import javax.persistence.LockModeType
+import javax.persistence.Persistence
+import javax.persistence.Query
+import spock.lang.Shared
+import spock.lang.Unroll
+
+class EntityManagerTest extends AbstractHibernateTest {
+
+  @Shared
+  EntityManagerFactory entityManagerFactory = Persistence.createEntityManagerFactory("test-pu")
+
+  @Unroll
+  def "test hibernate action #testName"() {
+    setup:
+
+    EntityManager entityManager = entityManagerFactory.createEntityManager()
+    EntityTransaction entityTransaction = entityManager.getTransaction()
+    entityTransaction.begin()
+
+    def entity = prepopulated.get(0)
+    if (attach) {
+      entity = runUnderTrace("setup") {
+        entityManager.merge(prepopulated.get(0))
+      }
+      ignoreTracesAndClear(1)
+    }
+    try {
+      sessionMethodTest.call(entityManager, entity)
+    } catch (Exception e) {
+      // We expected this, we should see the error field set on the span.
+    }
+
+    entityTransaction.commit()
+    entityManager.close()
+
+    expect:
+    boolean isPersistTest = "persist" == testName
+    assertTraces(1) {
+      trace(0, 4 + (isPersistTest ? 1 : 0)) {
+        span(0) {
+          name "Session"
+          kind INTERNAL
+          hasNoParent()
+          attributes {
+          }
+        }
+        span(1) {
+          name ~/Session.$methodName $resource/
+          kind INTERNAL
+          childOf span(0)
+          attributes {
+          }
+        }
+
+        def offset = 0
+        if (isPersistTest) {
+          // persist test has an extra query for getting id of inserted element
+          offset = 1
+          span(2) {
```

Comment:
Name of this span is a normalized sql query. I added a name for this one because it is used only by one test and always has the same value.
This test is based on a test from `SessionTest` which also does not assert names for sql queries presumably this is because the sql queries differ between testcases. To me asserting them doesn't really seem that important because this test is about hibernate instrumentation not jdbc. If you want I can add the asserts.

Reply:
The reason I asked for name is that without name it is hard to understand why all these span exist. This indeed is probably more about span modelling in general than this PR in particular.